src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -2512,3 +2496,76 @@ export function updateSavedUserPreferenceChartIds(
     }
     return chartSettings;
 }
+
+export function getMolecularProfileOptions(
+    molecularProfiles: MolecularProfile[],
+    molecularPorfileSampleSet: { [id: string]: string[] }
+): GenomicDataCountWithSampleUniqueKeys[] {
+    return _.chain(molecularProfiles)
+        .groupBy(molecularProfile =>
+            molecularProfile.molecularProfileId.replace(
+                molecularProfile.studyId + '_',
+                ''
+            )
+        )
+        .map((profiles, value) => {
+            const uniqueProfiledsamples = _.chain(profiles)
+                .flatMap(
+                    molecularProfile =>
+                        molecularPorfileSampleSet[
+                            molecularProfile.molecularProfileId
+                        ] || []
+                )
+                .uniq()
+                .value();
+            return {
+                value: value,
+                label: profiles[0].name,
+                count: uniqueProfiledsamples.length,
+                sampleUniqueKeys: uniqueProfiledsamples,
+            };
+        })
+        .value();
+}
+
+export function getCNASamplesCount(molecularPorfileSampleCountSet: {
+    [id: string]: number;
+}) {
+    let count = 0;
+    if (molecularPorfileSampleCountSet['cna'] !== undefined) {
+        count = molecularPorfileSampleCountSet['cna'];
+    } else if (molecularPorfileSampleCountSet['gistic'] !== undefined) {
+        count = molecularPorfileSampleCountSet['gistic'];
+    } else if (molecularPorfileSampleCountSet['rae'] !== undefined) {
+        count = molecularPorfileSampleCountSet['rae'];
+    } else if (molecularPorfileSampleCountSet['cna_consensus'] !== undefined) {
+        count = molecularPorfileSampleCountSet['cna_consensus'];
+    }
+    return count;
+}
+
+export function getMolecularPorfileSamplesSet(
+    samples: Sample[],
+    genePanelData: GenePanelData[]
+) {
+    const selectedUniqueSampleKeySet = _.keyBy(
```

Comment:
seems like from perspective of this function, there is no "selection."  it's just whatever samples are passed, no?